nni/tools/nnictl/common_utils.py

```diff
@@ -95,3 +98,43 @@ def generate_folder_name():
         temp_dir = generate_folder_name()
     os.makedirs(temp_dir)
     return temp_dir
+
+class SimplePreemptiveLock(filelock.SoftFileLock):
+    '''this is a lock support check lock expiration, if you do not need check expiration, you can use SoftFileLock'''
+    def __init__(self, lock_file, check_interval=-1):
+        super(__class__, self).__init__(lock_file, check_interval)
+        self._lock_file_name = '{}.{}'.format(self._lock_file, os.getpid())
+
+    def __enter__(self):
+        while True:
+            try:
+                self.acquire()
+                return self
```

Comment:
base class already has a loop, we can set `timeout=-1` for base class,  then we do not need this `__enter__`

Reply:
we can rename the `check_interval` back to `stale` with this new design.

Reply:
Indeed, we do not need `__enter__` any more, I will remove it.